piccolo/table.py

```diff
@@ -212,6 +211,14 @@ def __init_subclass__(
                 if isinstance(column, (JSON, JSONB)):
                     json_columns.append(column)
 
+        if not primary_key_column:
+            primary_key_column = cls._create_serial_primary_key()
+
+            columns.insert(0, primary_key_column)  # PK should be added first
+            default_columns.append(primary_key_column)
+
+        cls.id = primary_key_column
```

Comment:
There could be a subtle issue here. Imagine someone defines a primary key which isn't called `id` (for example `pk`). We would then end up with two primary key columns (`id` and `pk`), which would fail.

A nice solution would be to add a new attribute to `TableMeta` called `primary_key`. In the example above, we would assign the `pk` column to `TableMeta.primary_key`.  So rather than us always assuming the primary key is called `id`, we can access it by doing `SomeTable._meta.primary_key`.

We would effectively be killing two birds with one stone - the user can define the column type used as the primary key, and also the name of the column.

I don't think there are two many places which assume the primary key is called 'id'. Here are some example:

* https://github.com/piccolo-orm/piccolo/blob/master/piccolo/utils/sql_values.py#L21
* https://github.com/piccolo-orm/piccolo/blob/master/piccolo/table.py#L319
* https://github.com/piccolo-orm/piccolo/blob/master/piccolo/table.py#L334
* https://github.com/piccolo-orm/piccolo/blob/master/piccolo/table.py#L376
* https://github.com/piccolo-orm/piccolo/blob/master/piccolo/table.py#L415

What do you think? I can help implement these changes.

Reply:
I love that idea, I thought about that too. Let me give it a go tomorrow and if I see if it's too much I'll leave that to you.

Reply:
@dantownsend I've started this change but a dozen of tests are failing, mainly with `sqlite3.InterfaceError: Error binding parameter 0 - probably unsupported type.`
I've also added a test to define `pk` as `primary key` on table, it's passing but I have a feeling it's not right, it inserts 0 in `Serial`.
Any help with this would be great.

Reply:
I think the  `sqlite3.InterfaceError: Error binding parameter 0 - probably unsupported type.` error is being caused by this:

```python
.where(MyTable._meta.primary_key == row._meta.primary_key)
```

The problem is `row._meta.primary_key`. It's returning the column instance (e.g. Sequence()), rather than the column value (e.g. 1).

It's a bit untidy, but this works:

```python
.where(MyTable._meta.primary_key == getattr(row, MyTable._meta.primary_key._meta.name)
```